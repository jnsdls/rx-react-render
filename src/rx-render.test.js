import React from 'react';
import { from, BehaviorSubject } from 'rxjs';

import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { mount, shallow } from 'enzyme';

import RxReactRender from './rx-render';

Enzyme.configure({ adapter: new Adapter() });

describe('<RxReactRender />', () => {
  let wrapper = null;

  //   beforeEach(() => {
  //     const root = document.createElement('div');
  //     wrapper = mount(<RxReactRender />, { attachTo: root });
  //   });

  it('without props it renders the contents', () => {
    wrapper = shallow(<RxReactRender>{() => <div>No Props</div>}</RxReactRender>);
    expect(wrapper).toMatchSnapshot();
  });

  it('with non-observable props it renders with the props passed through', () => {
    wrapper = shallow(
      <RxReactRender notAnObservable="DefinitelyNotAnObservable">
        {({ notAnObservable }) => <div>{notAnObservable}</div>}
      </RxReactRender>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('with observable props it renders with the values passed through', () => {
    const observable = from(['this is just one thing']);
    wrapper = shallow(
      <RxReactRender observable={observable}>
        {({ observable }) => <div>Observable: {observable}</div>}
      </RxReactRender>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('with observable props it renders with the values passed through and updates work', () => {
    const observable = new BehaviorSubject('this is just one thing');
    wrapper = shallow(
      <RxReactRender observable={observable}>
        {({ observable }) => <div>Observable: {observable}</div>}
      </RxReactRender>
    );
    expect(wrapper).toMatchSnapshot();

    observable.next('the next thing');
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });

  it('handles any type of prop', () => {
    const observable = new BehaviorSubject('this is just one thing');
    wrapper = shallow(
      <RxReactRender notObservable={'just a string'} observable={observable}>
        {({ observable, notObservable }) => (
          <div>
            Observable: {observable} - Not An Observable: {notObservable}
          </div>
        )}
      </RxReactRender>
    );
    expect(wrapper).toMatchSnapshot();

    observable.next('the next thing');
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });
});
