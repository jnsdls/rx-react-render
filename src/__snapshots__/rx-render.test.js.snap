// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RxReactRender /> handles any type of prop 1`] = `
<div>
  Observable: 
  this is just one thing
   - Not An Observable: 
  just a string
</div>
`;

exports[`<RxReactRender /> handles any type of prop 2`] = `
<div>
  Observable: 
  the next thing
   - Not An Observable: 
  just a string
</div>
`;

exports[`<RxReactRender /> with non-observable props it renders with the props passed through 1`] = `
<div>
  DefinitelyNotAnObservable
</div>
`;

exports[`<RxReactRender /> with observable props it renders with the values passed through 1`] = `
<div>
  Observable: 
  this is just one thing
</div>
`;

exports[`<RxReactRender /> with observable props it renders with the values passed through and updates work 1`] = `
<div>
  Observable: 
  this is just one thing
</div>
`;

exports[`<RxReactRender /> with observable props it renders with the values passed through and updates work 2`] = `
<div>
  Observable: 
  the next thing
</div>
`;

exports[`<RxReactRender /> without props it renders the contents 1`] = `
<div>
  No Props
</div>
`;
